### PR TITLE
Check ECC-CDH is compliant with SP800-56A-r3

### PR DESCRIPTION
**Pull request by slontis**
Added comments and cleared an intermediate result.
KAT tests already exist in evppkey.txt (Search for "KAS_ECC_CDH_PrimitiveTest")

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/ec/ecdh_ossl.c [hunk 1]
```diff
@@ -83,11 +92,20 @@ int ecdh_simple_compute_key(unsigned char **pout, size_t *poutlen,
         goto err;
     }
 
+    /*
+     * Step(2) :If point tmp is at affinity then clear intermediate values and
```
Comment (mattcaswell) [inline crypto/ec/ecdh_ossl.c]: Err...I think you meant "at infinity"!

File: crypto/ec/ecdh_ossl.c [hunk 2]
```diff
@@ -83,11 +92,20 @@ int ecdh_simple_compute_key(unsigned char **pout, size_t *poutlen,
...
+     * exit.Note: getting affine coordinates returns 0 if point is at infinity.
```
Comment (mattcaswell) [inline crypto/ec/ecdh_ossl.c]: Nit: space after "exit."

**Comment by slontis** [comment]
updated nits.

**Comment by slontis** [comment]
Thanks. Merged to master.

**Pull request closed without merging**